sdk/tests/core/mock_az_mqtt.c

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Contains the az_mqtt.h mock implementation.
+ *
+ */
+
+#include <azure/core/az_mqtt.h>
+#include <azure/core/az_mqtt_config.h>
+#include <azure/core/az_platform.h>
+#include <azure/core/az_span.h>
+#include <azure/core/internal/az_log_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/core/internal/az_result_internal.h>
+#include <azure/core/internal/az_span_internal.h>
+
+#include <mosquitto.h>
+#include <pthread.h>
+#include <stdlib.h>
+
+#include <azure/core/_az_cfg.h>
+
+#ifdef _az_MOCK_ENABLED
```

Comment:
Should this surround the #includes too?

Reply:
+1 - please check with various CMake options to ensure we can still build with or without mocks enabled.

Reply:
I removed the `_az_MOCK_ENABLED`. Instead, if `_az_MOCK_ENABLED` is not set, this file will not compile